simpleclient_logback/src/main/java/io/prometheus/client/logback/InstrumentedAppender.java

```diff
@@ -0,0 +1,84 @@
+package io.prometheus.client.logback;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+
+public class InstrumentedAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+
+  public static final String COUNTER_NAME = "logback_appender_total";
+
+  private final CollectorRegistry registry;
+
+  private Counter counter;
+  private Counter.Child trace;
+  private Counter.Child debug;
+  private Counter.Child info;
+  private Counter.Child warn;
+  private Counter.Child error;
+
+  /**
+   * Create a new instrumented appender using the default registry.
+   */
+  public InstrumentedAppender() {
+    this(CollectorRegistry.defaultRegistry);
+  }
+
+  /**
+   * Create a new instrumented appender using the given registry.
+   *
+   * @param registry the collector registry
+   */
+  public InstrumentedAppender(CollectorRegistry registry) {
+    this.registry = registry;
+    setName(Appender.class.toString());
+  }
+
+  @Override
+  public void start() {
```

Comment:
Yeah, multiple instances of the same appender can be declared in the logging config but in this case that should thrown an exception as we would double count otherwise seeing as we only ever have a single registry. We can have a static boolean to check and throw this exception explicitly if we want. A static counter would have hidden that issue and led to so!e fun debugging! Typically each appender would have to have different key config such as filename. This doesn't apply here - different registries is only use case I can think of and statics again couldn't work for that.

AppenderBase/Skeleton just do the work of ensuring its started / initialised before doAppend is called and in the logback impl we made it so its not sync doAppend as we don't need it to be. 

I don't mind if you want to update the pr to use statics. I consider it bad coding style for this usage though so I'd rather not add them myself.